common/sagas/deterministicWallets.js

```diff
@@ -62,16 +72,22 @@ function* getDeterministicWallets(
 function* updateWalletValues(): Generator<Yield, Return, Next> {
   const node: BaseNode = yield select(getNodeLib);
   const wallets: DeterministicWalletData[] = yield select(getWallets);
-  const calls = wallets.map(w => apply(node, node.getBalance, [w.address]));
-  const balances = yield all(calls);
-
-  for (let i = 0; i < wallets.length; i++) {
-    yield put(
-      updateDeterministicWallet({
-        ...wallets[i],
-        value: balances[i]
-      })
-    );
+
+  try {
+    const calls = wallets.map(w => apply(node, node.getBalance, [w.address]));
+    const balances = yield all(calls);
+
+    for (let i = 0; i < wallets.length; i++) {
+      yield put(
+        updateDeterministicWallet({
+          ...wallets[i],
+          value: balances[i]
+        })
+      );
+    }
+  } catch (err) {
+    console.log(err);
+    //TODO: maybe communicate API call error to user?
```

Comment:
Definitely, I think you can just get away with a `yield put(showNotification(...`